packages/nimble-components/src/table-column/enum-text/tests/table-column-enum-text.spec.ts

```diff
@@ -0,0 +1,547 @@
+            });
+        }
+    });
+
+    it('displays blank when no matches', async () => {
+        ({ element, connect, disconnect } = await setup([
+            { key: 'a', label: 'alpha' }
+        ]));
+        pageObject = new TablePageObject<SimpleTableRecord>(element);
+        await element.setData([{ field1: 'no match' }]);
+        await connect();
+        await waitForUpdatesAsync();
+
+        expect(pageObject.getRenderedCellContent(0, 0)).toBe('');
+    });
+
+    it('displays default when no other matches', async () => {
+        ({ element, connect, disconnect } = await setup([
+            { key: 'a', label: 'alpha' },
+            { defaultMapping: true, label: 'bravo' }
+        ]));
+        pageObject = new TablePageObject<SimpleTableRecord>(element);
+        await element.setData([{ field1: 'no match' }]);
+        await connect();
+        await waitForUpdatesAsync();
+
+        expect(pageObject.getRenderedCellContent(0, 0)).toBe('bravo');
+    });
+
+    it('changing fieldName updates display', async () => {
+        ({ element, connect, disconnect } = await setup([
+            { key: 'a', label: 'alpha' },
+            { key: 'b', label: 'bravo' }
+        ]));
+        pageObject = new TablePageObject<SimpleTableRecord>(element);
+        await element.setData([{ field1: 'a', field2: 'b' }]);
+        await connect();
+        await waitForUpdatesAsync();
+
+        const firstColumn = element.columns[0] as TableColumnEnumText;
+        firstColumn.fieldName = 'field2';
+        await waitForUpdatesAsync();
+
+        expect(pageObject.getRenderedCellContent(0, 0)).toBe('bravo');
+    });
+
+    it('changing mapping label updates display', async () => {
+        ({ element, connect, disconnect } = await setup([
+            { key: 'a', label: 'alpha' }
+        ]));
+        pageObject = new TablePageObject<SimpleTableRecord>(element);
+        await element.setData([{ field1: 'a' }]);
+        await connect();
+        await waitForUpdatesAsync();
+
+        const mapping = (element.columns[0] as TableColumnEnumText)
+            .mappings[0] as MappingText;
+        mapping.label = 'bravo';
+        await waitForUpdatesAsync();
+
+        expect(pageObject.getRenderedCellContent(0, 0)).toBe('bravo');
+    });
+
+    it('changing mapping key updates display', async () => {
+        ({ element, connect, disconnect } = await setup([
+            { key: 'a', label: 'alpha' }
+        ]));
+        pageObject = new TablePageObject<SimpleTableRecord>(element);
+        await element.setData([{ field1: 'b' }]);
+        await connect();
+        await waitForUpdatesAsync();
+
+        const mapping = (element.columns[0] as TableColumnEnumText)
+            .mappings[0] as MappingText;
+        mapping.key = 'b';
+        await waitForUpdatesAsync();
+
+        expect(pageObject.getRenderedCellContent(0, 0)).toBe('alpha');
+    });
+
+    it('sets title when cell text is ellipsized', async () => {
```

Comment:
Since we're using the base class I'd argue that these ellipsize and title tests aren't really necessary; we can rely on them being covered by the text column.

Reply:
Ok, removed.